pkgs/tools/misc/dialog/default.nix

```diff
@@ -3,27 +3,30 @@
 , fetchurl
 , libtool
 , ncurses
-, withLibrary ? false
-, unicodeSupport ? true
 , enableShared ? !stdenv.isDarwin
+, unicodeSupport ? true
+, withLibrary ? false
 }:
 
-assert withLibrary -> libtool != null;
-assert unicodeSupport -> ncurses != null && ncurses.unicodeSupport;
+assert unicodeSupport -> ncurses.unicodeSupport;
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "dialog";
-  version = "1.3-20220728";
+  version = "1.3-20230209";
 
   src = fetchurl {
-    url = "ftp://ftp.invisible-island.net/dialog/dialog-${finalAttrs.version}.tgz";
-    hash = "sha256-VEGJc9VZpGGwBpX6/mjfYvK8c9UGtDaCHXfKPfRUGQs=";
+    url = "https://invisible-island.net/archives/dialog/dialog-${finalAttrs.version}.tgz";
```

Comment:
I don't see how using `finalAttrs` here helps with anything. In case of an override, the hash will be wrong anyways requiring overwriting the entire `src` Attribute anyhow. Am I wrong? Is there an interface to change the `hash` attribute after the call to `fetchurl`.

Reply:
I use `finalAttrs` because I don't like `rec` (https://nix.dev/recipes/best-practices#recursive-attribute-set-rec).

Reply:
Yes, that makes a lot of sense.